File: data/transcoder_evaluation_gfg/go/FIBONACCI_MODULO_P.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(p int) int {
	var (
		first  int = 1
		second int = 1
		number int = 2
		next   int = 1
	)
	for next != 0 {
		next = (first + second) % p
		first = second
		second = next
		number++
	}
	return number
}
//TOFILL
func main() {
	var (
		n_success int     = 0
		param0    []int = []int{51, 40, 68, 7, 8, 32, 93, 75, 71, 15}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i]) == f_gold(param0[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/rust/FIBONACCI_MODULO_P.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut p: i32) -> i32 {
    let mut first: i32 = 1 as i32;
    let mut second: i32 = 1 as i32;
    let mut number: i32 = 2 as i32;
    let mut next: i32 = 1 as i32;
    while next != 0 {
        next = (first + second) % p;
        first = second;
        second = next;
        number += 1
    }
    return number;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0: [i32; 10] =
        [51 as i32, 40 as i32, 68 as i32,
         7 as i32, 8 as i32, 32 as i32,
         93 as i32, 75 as i32, 71 as i32,
         15 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize]) == f_gold(param0[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
